cmd/init.go

```diff
@@ -0,0 +1,89 @@
+package cmd
+
+import (
+	"context"
+	"io/ioutil"
+
+	"github.com/pastelnetwork/pastel-utility/configs"
+	"github.com/pastelnetwork/gonode/common/cli"
+	"github.com/pastelnetwork/gonode/common/log"
+	"github.com/pastelnetwork/gonode/common/log/hooks"
+	"github.com/pastelnetwork/gonode/common/sys"
+	"github.com/pkg/errors"
+)
+
+func setupInitCommand(app *cli.App, config *configs.Config) {
+	// define flags here
+	var workDirectoryFlag string
+
+	initCommand := cli.NewCommand("init")
+	// initCommand.CustomAppHelpTemplate = getColoredHeaders(cyan)
+	initCommand.CustomHelpTemplate = getColoredHeaders(cyan)
+	initCommand.SetUsage("Command that performs initialization of the system for both Wallet and SuperNodes")
+	initCommandFlags := []*cli.Flag{
+		cli.NewFlag("work-dir", &workDirectoryFlag),
+	}
+	initCommand.AddFlags(initCommandFlags...)
+	addLogFlags(initCommand, config)
+
+	// create walletnode and supernode subcommands
+	walletnodeSubCommandName := green.Sprint("walletnode")
+	walletnodeSubCommand := cli.NewCommand(walletnodeSubCommandName)
+	walletnodeSubCommand.Usage = cyan.Sprint("Perform wallet specific initialization after common")
+
+	supernodeSubCommandName := green.Sprint("supernode")
+	supernodeSubCommand := cli.NewCommand(supernodeSubCommandName)
+	supernodeSubCommand.Usage = cyan.Sprint("Perform Supernode/Masternode specific initialization after common")
+	initSubCommands := []*cli.Command{
+		walletnodeSubCommand,
+		supernodeSubCommand,
+	}
+	initCommand.AddSubcommands(initSubCommands...)
+
+	initCommand.SetActionFunc(func(ctx context.Context, args []string) error {
+		ctx = log.ContextWithPrefix(ctx, "app")
+
+		if config.Quiet {
+			log.SetOutput(ioutil.Discard)
+		} else {
+			log.SetOutput(app.Writer)
+		}
+
+		if config.LogFile != "" {
+			fileHook := hooks.NewFileHook(config.LogFile)
+			log.AddHook(fileHook)
+		}
+
+		if err := log.SetLevelName(config.LogLevel); err != nil {
+			return errors.Errorf("--log-level %q, %v", config.LogLevel, err)
+		}
+
+		log.Info("flag-name: ", workDirectoryFlag)
+
+		return runInit(ctx, config)
+	})
```

Comment:
Why do you repeat this code in every command? These are global flags.

pastel-utility [global args] [command] [command args]